src/src/PluginStructs/P104_data_struct.cpp

```diff
@@ -33,13 +33,31 @@ P104_data_struct::P104_data_struct(MD_MAX72XX::moduleType_t _mod,
                                    uint8_t                  _modules,
                                    uint8_t                  _zonesCount)
   : mod(_mod), taskIndex(_taskIndex), cs_pin(_cs_pin), modules(_modules), expectedZones(_zonesCount) {
-  if (Settings.InitSPI > 0) { // FIXME tonhuisman: Replace by isSPI_valid() once merged from PR #3773
-    P = new MD_Parola(mod, cs_pin, modules);
+  if (Settings.isSPI_valid()) {
+    P = new (std::nothrow) MD_Parola(mod, cs_pin, modules);
   } else {
     addLog(LOG_LEVEL_ERROR, F("DOTMATRIX: Required SPI not enabled. Initialization aborted!"));
   }
 }
 
+/*******************************
+ * Destructor
+ ******************************/
+P104_data_struct::~P104_data_struct() {
+  # ifdef P104_USE_BAR_GRAPH
+
+  if (nullptr != pM) {
+    pM = nullptr; // Not created here, only reset
+  }
+  # endif // ifdef P104_USE_BAR_GRAPH
+
+  if (nullptr != P) {
+    // do not: delete P; // Warning: the MD_Parola object doesn't have a virtual destructor, and when changed,
+    // a reboot uccurs when the object is deleted here!
+    P = nullptr; // Reset only
```

Comment:
memory leak !
Can this object be shared?
if so, then make it static.
If not then add virtual destructor or maybe use it as a member in the data struct?

Reply:
See my remark above (unresolved it).
It can't be shared, so making it static isn't an option. I'll investigate if I can initialize it differently.

Reply:
If it can't be shared and destruction is also difficult, you may want to declare a static array of pointer of that type.
Array length = max. nr of tasks.